src/client/invite/Invite.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import { Form, Input } from 'antd';
+import Affix from '../components/Utils/Affix';
+import LeftSidebar from '../app/Sidebar/LeftSidebar';
+import requiresLogin from '../auth/requiresLogin';
+import { getAuthenticatedUserName } from '../reducers';
+import './Invite.less';
+
+@requiresLogin
+@connect(
+  state => ({
+    authenticatedUserName: getAuthenticatedUserName(state),
+  }),
+)
+export default class Invite extends React.Component {
+  static propTypes = {
+    authenticatedUserName: PropTypes.string,
+  };
+
+  static defaultProps = {
+    authenticatedUserName: '',
+  };
+
+  render() {
+    const { authenticatedUserName } = this.props;
+    const port = window.location.port ? `:${window.location.port}` : '';
```

Comment:
Can we use `location` from react-router? `window` is undefined on the server.

Reply:
react router `location` give me only `pathname`, i can't get `port`, `host`, or `protocol` from it. But i can check for window availability before generating the link.